compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -1090,25 +1119,73 @@ abstract class PrepJSInterop extends plugins.PluginComponent
           "Classes, traits and objects not extending js.Any may not have an " +
           "@js.native annotation")
     } else {
-      checkJSNativeSpecificAnnotsOnNonJSNative(pos, sym)
+      checkJSNativeSpecificAnnotsOnNonJSNative(sym)
     }
   }
 
-  private def checkJSNativeSpecificAnnotsOnNonJSNative(pos: Position,
-      sym: Symbol): Unit = {
-    if (sym.hasAnnotation(JSNameAnnotation)) {
-      reporter.warning(pos,
-          "Non JS-native classes, traits and objects should not have an " +
-          "@JSName annotation, as it does not have any effect. " +
-          "This will be enforced in 1.0.")
+  private def checkJSNativeSpecificAnnotsOnNonJSNative(sym: Symbol): Unit = {
+    for (annot <- sym.annotations) {
+      if (annot.symbol == JSNameAnnotation) {
+        reporter.warning(annot.pos,
+            "Non JS-native classes, traits and objects should not have an " +
+            "@JSName annotation, as it does not have any effect. " +
+            "This will be enforced in 1.0.")
+      } else if (annot.symbol == JSImportAnnotation) {
+        reporter.error(annot.pos,
+            "Non JS-native classes, traits and objects may not have an " +
+            "@JSImport annotation.")
+      } else if (annot.symbol == JSGlobalScopeAnnotation) {
+        reporter.error(annot.pos,
+            "Only native JS objects can have an @JSGlobalScope annotation " +
+            "(or extend js.GlobalScope).")
+      }
     }
-    if (sym.hasAnnotation(JSImportAnnotation)) {
-      reporter.error(pos,
-          "Non JS-native classes, traits and objects may not have an " +
-          "@JSImport annotation")
+  }
+
+  private def checkAndGetJSNativeLoadingSpecAnnotOf(pos: Position,
+      sym: Symbol): Option[Annotation] = {
+    val annots = sym.annotations.filter {
+      annot => JSNativeLoadingSpecAnnots.contains(annot.symbol)
```

Comment:
Is this new, that you write the argument on the new line? Examples in CODINGSTYLE are different.